main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService.cs

```
@@ -774,6 +774,17 @@ public MonoDevelop.Projects.Project GetMonoProject (Microsoft.CodeAnalysis.Docum
 
 		public static Func<Task> FreezeLoad = () => Task.CompletedTask;
 
+		internal static async Task SafeFreezeLoad ()
+		{
+			try {
+				await FreezeLoad ();
+			} catch (Exception) {
```

We might want to LogInternalError on Exception and swallow just OperationCanceledException

---

Not sure it is worth it. Any non-cancel exception here would be caught by the NuGet addin.

---

Then we should filter OperationCanceledException only. General catch exception is something we want to eliminate.

---

If this freeze NuGet restore task throws an exception (not sure that it will) then it would stop the type system from continuing if there is no general catch exception. The NuGet addin will log exceptions itself but the type system would still receive the task's non-cancelled exception. The type system does not care about what happens when the freeze task finishes (either successfully or is cancelled or an error occurs) - it is just used to prevent it from running when the freeze task is active. So I am not sure we should remove the general catch exception here.

---

Fine by me then.